data/projects/optimize_caching.md

```diff
@@ -9,20 +9,28 @@ Own performance optimization ideas are highly encouraged!
 
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
 * The applicant has determined a number of circumstances where coala needlessly
 checks unmodified files.
 * The applicant has created testcases to validate performance improvements
 * A cEP describing the changes is merged.
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
 * The applicant has achieved a measurable performance benefit in at least one
 of the above cases.
 * The respective code has been properly tested, documented and merged.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
-* At least two more cases have been 'solved' in a similar fashion.
+* Documentation and code for the 'section division' for 'per-section caching' has been completed and merged.
+* Code for Git like approach to not explore directories that haven't been modified has been tested and merged.
+* Own idea for performance optimization(If any) has been properly documented and work has been started. 
+  
```

Comment:
In preparation and phase 1 no specific methods were mentioned and the student suddenly has to implement exactly these things. These need to be mentioned earlier. Maybe also Update the text above the milestones as well as the bonding and phase 1 milestones as well?